files/build_templates/docker_image_ctl.j2

```diff
@@ -300,6 +306,14 @@ function postStartAction()
     REDIS_BMP_SOCK="/var/run/redis/redis_bmp.sock"
     chgrp -f redis $REDIS_SOCK && chmod -f 0760 $REDIS_SOCK
     chgrp -f redis $REDIS_BMP_SOCK && chmod -f 0760 $REDIS_BMP_SOCK
+
+    if [[ $DEV && $midplane_ip ]]; then
+        IFS=_ read ip port < <(jq -r '.INSTANCES | [.redis.hostname, .redis.port] | join("_")' /var/run/redis$DEV/sonic-db/database_config.json)
```

Comment:
Instead doing a bind here using redis-cli. Is it possible to update the `hostIP` to `midplane-ip` database_config.json for chassis?

Reply:
I have explored that possibility as well.

The issue is that midplane IPs take a little longer to get assigned when the switch is coming up whereas the docker network IPs are already available by the time this script starts running: https://github.com/sonic-net/sonic-buildimage/blob/master/dockers/docker-database/docker-database-init.sh#L5-L10

If we want to update database_config.json and bind redis instances to midplane IPs alone, the instantiation of redis database instances have to be delayed till the midplane IPs are assigned. 

Delaying something critical as database can have consequences and it's an unknown territory. We don't know all the consequences it can have.